python/ray/dag/compiled_dag_node.py

```diff
@@ -1425,23 +1476,35 @@ def _generate_dag_operation_graph_node(
                 # and WRITE. Each _DAGOperationGraphNode has a _DAGNodeOperation.
                 task_index = exec_task.task_idx
                 dag_node = self.idx_to_task[task_index].dag_node
+                method_name = exec_task.method_name
                 actor_handle = dag_node._get_actor_handle()
                 requires_nccl = dag_node.type_hint.requires_nccl()
+                upstream_requires_nccl = False
```

Comment:
Why this change?

Reply:
We need to pass `requires_nccl` to `_DAGOperationGraphNode` constructor. For read op, originally it uses current DAGNode's `requires_nccl` which is not accurate (which means if the write op requires nccl). Instead we should check if the upstream node requires nccl.